addons/im_support/static/tests/systray_tests.js

```diff
@@ -0,0 +1,342 @@
+        'cors: /odoo_im_support/fetch_messages',
+        'cors: /odoo_im_support/chat_post',
+    ]);
+
+    messagingMenu.destroy();
+});
+
+QUnit.test('fold Support channel', function (assert) {
+    assert.expect(11);
+
+    var messagingMenu = new systray.MessagingMenu();
+    addMockSupportEnvironment(messagingMenu, {
+        mockRPC: function (route, args) {
+            assert.step(args.method || route);
+            return this._super.apply(this, arguments);
+        },
+        mockSupportRPC: function (route) {
+            if (route.split('/')[1] === 'odoo_im_support') {
+                assert.step('cors: ' + route);
+            }
+            return this._super.apply(this, arguments);
+        },
+        services: this.services,
+        session: this.supportParams,
+    });
+    testUtils.intercept(messagingMenu, 'call_service', function (ev) {
+        if (ev.data.service === 'local_storage') {
+            assert.step('LocalStorage: ' + ev.data.method + ' ' + ev.data.args);
+        }
+    }, true);
+    messagingMenu.appendTo($('#qunit-fixture'));
+
+    messagingMenu.$('.dropdown-toggle').click();
+    assert.strictEqual(messagingMenu.$('.o_mail_channel_preview[data-channel_id=SupportChannel]').length,
+        1, "should display the Support channel");
+
+    messagingMenu.$('.o_mail_channel_preview[data-channel_id=SupportChannel]').click();
+    assert.strictEqual($('.o_chat_window').length, 1,
+        "should have open a chat window");
+
+    // fold, re-open and close channel
+    $('.o_chat_window .o_chat_title').click();
+    $('.o_chat_window .o_chat_title').click();
+    $('.o_chat_window .o_chat_window_close').click();
+
+    assert.verifySteps([
+        '/mail/init_messaging',
+        'message_fetch',
+        'cors: /odoo_im_support/get_support_channel',
+        'LocalStorage: setItem im_support.channel_state,open',
+        'cors: /odoo_im_support/fetch_messages',
+        'LocalStorage: setItem im_support.channel_state,folded',
+        'LocalStorage: setItem im_support.channel_state,open',
+        'LocalStorage: setItem im_support.channel_state,closed',
+    ]);
+
+    messagingMenu.destroy();
+});
+
+QUnit.test('restore Support channel if necessary', function (assert) {
+    assert.expect(5);
+
+    var messagingMenu = new systray.MessagingMenu();
+    addMockSupportEnvironment(messagingMenu, {
+        mockRPC: function (route, args) {
+            assert.step(args.method || route);
+            return this._super.apply(this, arguments);
+        },
+        mockSupportRPC: function (route) {
+            if (route.split('/')[1] === 'odoo_im_support') {
+                assert.step('cors: ' + route);
+            }
+            return this._super.apply(this, arguments);
+        },
+        services: this.services,
+        session: this.supportParams,
+    });
+    testUtils.intercept(messagingMenu, 'call_service', function (ev) {
+        if (ev.data.service === 'local_storage') {
```

Comment:
i see that this is an example of something that should be easily done in a test util function.